osu.Game.Rulesets.Osu/Mods/OsuModFlashlight.cs

```diff
@@ -1,31 +1,58 @@
 ﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
 // See the LICENCE file in the repository root for full licence text.
 
+using System.Collections.Generic;
+using System.Linq;
 using osu.Framework.Bindables;
 using osu.Framework.Graphics;
 using osu.Framework.Input;
 using osu.Framework.Input.Events;
 using osu.Game.Rulesets.Mods;
+using osu.Game.Rulesets.Objects.Drawables;
 using osu.Game.Rulesets.Osu.Objects;
+using osu.Game.Rulesets.Osu.Objects.Drawables;
 using osuTK;
 
 namespace osu.Game.Rulesets.Osu.Mods
 {
-    public class OsuModFlashlight : ModFlashlight<OsuHitObject>
+    public class OsuModFlashlight : ModFlashlight<OsuHitObject>, IApplicableToDrawableHitObjects
     {
         public override double ScoreMultiplier => 1.12;
 
         private const float default_flashlight_size = 180;
 
-        public override Flashlight CreateFlashlight() => new OsuFlashlight();
+        private OsuFlashlight flashlight;
+
+        public override Flashlight CreateFlashlight() => flashlight = new OsuFlashlight();
+
+        public void ApplyToDrawableHitObjects(IEnumerable<DrawableHitObject> drawables)
+        {
+            foreach (DrawableSlider drawable in drawables.OfType<DrawableSlider>())
+            {
+                drawable.Tracking.ValueChanged += flashlight.OnSliderTrackingChange;
+            }
+        }
 
         private class OsuFlashlight : Flashlight, IRequireHighFrequencyMousePosition
         {
+            private int trackingSliders;
+
             public OsuFlashlight()
             {
                 FlashlightSize = new Vector2(0, getSizeFor(0));
             }
 
+            public void OnSliderTrackingChange(ValueChangedEvent<bool> e)
+            {
+                if (e.NewValue)
+                    trackingSliders++;
+                else
+                    trackingSliders--;
+
+                // If there are any sliders in a tracking state, apply a dim to the entire playfield over a brief duration.
+                this.TransformTo(nameof(FlashlightDim), trackingSliders > 0 ? 0.8f : 0.0f, 50);
```

Comment:
this should not run unless the count has changed to 0 or 1, for correctness. else you will interrupt an existing transition with a new duration.